app/services/fact_importer.rb

```diff
@@ -4,22 +4,28 @@ class FactImporter
   delegate :logger, :to => :Rails
   attr_reader :counters
 
+  def deprecation_warning
+    Foreman::Deprecation.deprecation_warning('2.0', 'Use Importer Registry to register or obtain facts for a plugin')
+  end
+
   def self.importer_for(type)
-    importers[type.to_s]
+    deprecation_warning
+    @importers.get(type)
   end
 
   def self.importers
-    @importers ||= {}.with_indifferent_access
+    deprecation_warning
+    @importers ||= Foreman::Plugin::ImporterRegistry.new
```

Comment:
This should use the existing registry, not create a new one:
```suggestion
    Foreman::Plugin.fact_registry.importers
```